internal/authproxy/serve.go

```diff
@@ -132,6 +144,37 @@ func Command() (*cobra.Command, error) {
 	return serveCmd, nil
 }
 
+type (
+	RequestStats struct {
+		LastRequest time.Time
+		mutex       sync.RWMutex
+	}
+)
+
+func NewStats() *RequestStats {
+	return &RequestStats{
+		LastRequest: time.Now(),
+	}
+}
+
+func (l *RequestStats) Process(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+
+		if err := next(c); err != nil {
+			c.Error(err)
```

Comment:
question: I think you can simply return the error, right? I dont see why you have to wrap it like that. I think the central error handler in echo will handle things properly if you just return the error.

```suggestion
			return err
```

Reply:
I don't really know, I was following the [middleware cookbook](https://echo.labstack.com/docs/cookbook/middleware)

Reply:
`Error` calls the HttpErrorHandler method so I think it's mainly about handling `echo` related errors that are not related to the error that might be returned by the rest of the code.